pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -240,65 +249,66 @@ func (h *calicoIPPoolHandler) updateROKSCalicoCfg(ctx context.Context) error {
 		return nil
 	}
 
-	// platform is ROKS, make sure that IPIPMode of default IPPool is Always
-
-	err = util.Update(ctx, h.iPPoolResourceInterface(), &calicoapi.IPPool{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: DefaultV4IPPoolName,
-		},
-	}, func(existing *calicoapi.IPPool) (*calicoapi.IPPool, error) {
-		if existing.Spec.IPIPMode == calicoapi.IPIPModeAlways {
-			logger.Infof("IPIPMode of %s IPPool already set to Always", DefaultV4IPPoolName)
-			return existing, nil // no need to update
-		}
-
-		if existing.Labels == nil {
-			existing.Labels = make(map[string]string)
-		}
-
-		if existing.Annotations == nil {
-			existing.Annotations = make(map[string]string)
-		}
-
-		// mark 'submariner-managed' label for persistency
-		existing.Labels[submarinerManagedLabel] = "true"
-		existing.Annotations[submarinerPrevIPIPMode] = string(existing.Spec.IPIPMode)
-		existing.Spec.IPIPMode = calicoapi.IPIPModeAlways
-
-		logger.Infof("Setting IPIPMode of %s IPPool to Always", DefaultV4IPPoolName)
-
-		return existing, nil
-	})
-
-	return errors.Wrap(err, "error updating default ippool")
+	// platform is ROKS, make sure that encapsulation of default Installation is set to IPIP
+	// In this way the Tigera Operator will set IPIPMode of default IPPool to Always
+	return util.Update(ctx, resource.ForDynamic(h.dynClient.Resource(InstallationsGVR)),
+		resource.MustToUnstructured(&tigerav1.Installation{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: DefaultInstallationName,
+			},
+		}), func(existing *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+			installation := &tigerav1.Installation{}
+			err := runtime.DefaultUnstructuredConverter.FromUnstructured(existing.Object, installation)
+			utilruntime.Must(err)
+
+			if installation.Spec.CalicoNetwork == nil {
+				installation.Spec.CalicoNetwork = &tigerav1.CalicoNetworkSpec{}
+			}
+
+			ipPools := installation.Spec.CalicoNetwork.IPPools
+			if len(ipPools) > 0 && ipPools[0].Encapsulation == tigerav1.EncapsulationIPIP {
+				logger.Infof("Encapsulation of " + DefaultInstallationName + " Installations already set to IPIP")
+				return existing, nil
+			}
+
+			if installation.Annotations == nil {
+				installation.Annotations = map[string]string{}
+			}
+
+			installation.Annotations[submarinerPrevEncapsulation] = ipPools[0].Encapsulation.String()
+			installation.Annotations[submarinerManagedLabel] = "true"
+
+			if len(ipPools) > 0 {
+				ipPools[0].Encapsulation = tigerav1.EncapsulationIPIP
+			} // Should add an IPPool if none exists?
```

Comment:
Sorry, just one more thing here... I had added this comment in my suggestion to raise awareness. If no pool exists and we don't need to add one, then we should also skip adding the annotations as well, so above:

```
            ipPools := installation.Spec.CalicoNetwork.IPPools
			if len(ipPools) == 0 {
                 logger.Infof("No IPPool found for Installation %q", DefaultInstallationName)
				return existing, nil
             } else if ipPools[0].Encapsulation == tigerav1.EncapsulationIPIP {
				logger.Infof("Encapsulation of %q Installation already set to IPIP", DefaultInstallationName)
				return existing, nil
			}
```

```suggestion
		    ipPools[0].Encapsulation = tigerav1.EncapsulationIPIP
```

Reply:
Changed.